### PR TITLE
Update golang and packages.

### PR DESCRIPTION
**Pull request by metanerd**
#### Summary
- updates golang version
- removes errors package dependency
- removes duplicate config files
- updates packages

Motivation: Updating go-github package to maybe reduce the rate limit of the requests.

### CODE REVIEW

File: server/builds.go [hunk 1]
```diff
@@ -33,12 +32,12 @@ func (b *Builds) getInstallationVersion(pr *model.PullRequest) string {
 
 func (b *Builds) dockerRegistryClient(s *Server) (reg *registry.Registry, err error) {
 	if _, err = url.ParseRequestURI(s.Config.DockerRegistryURL); err != nil {
-		return nil, errors.Wrap(err, "invalid url for docker registry")
+		return nil, fmt.Errorf("invalid url for docker registry")
```
Comment (hanzei) [inline server/builds.go:35]: There is a general consensus on using `github.com/pkg/errors` for error handling and wrapping. It's not documented jet, [but on it's way](https://github.com/mattermost/mattermost-developer-documentation/issues/612). Given that this codebase uses `github.com/pkg/errors` already a lot, I'm wondering why there is the need to move away from it.

Comment (metanerd) [reply server/builds.go:35]: Yeah, maybe I am too optimistic that go2 will be released soon. I will revert this.

File: server/builds.go [hunk 2]
```diff
@@ -207,7 +206,7 @@ func (b *Builds) checkBuildLink(ctx context.Context, s *Server, pr *model.PullRe
 		select {
 		case <-ctx.Done():
 			s.sendGitHubComment(pr.RepoOwner, pr.RepoName, pr.Number, "Timed out waiting for build link. Please check the logs.")
-			return "", fmt.Errorf("timed out waiting the build link")
+			return "", errors.Errorf("timed out waiting the build link")
```
Comment (hanzei) [inline server/builds.go:209]: Minor nit pick: `Errorf` is not needed here, `New` is enough
```suggestion
			return "", errors.New("timed out waiting the build link")
```

**Comment by hanzei** [comment]
@metanerd Is it intended that new commit invalidate reviews?

**Comment by hanzei** [comment]
@metanerd Is it intended that new commit invalidate reviews?
Screenshot from 2020-05-13 11-31-27

**Comment by metanerd** [comment]
> @metanerd Is it intended that new commit invalidate reviews?
> Screenshot from 2020-05-13 11-31-27

No, I will remove it.

**Pull request merged**